conan/api/subapi/command.py

```diff
@@ -20,5 +21,18 @@ def run(self, cmd):
             command = commands[current_cmd]
         except KeyError:
             raise ConanException(f"Command {current_cmd} does not exist")
+        # Conan has some global state in the ConanOutput class that
+        # get redefined when running a command and leak to the calling scope
+        # if running from a custom command.
+        # Store the old one and restore it after the command execution as a workaround.
+        _conan_output_level = ConanOutput._conan_output_level
```

Comment:
This pattern is spread thru the code in a few places, we could have something like this maybe?

```
@contenxtmanager
def restore_after(entity, *attributes):
    old_values = [getvalue(entity, attribute) for attribute in attributes)
    try:
        yield
    finally:
        for attribute, old_value in zip(attributes, old_values):
            setattr(entity, attribute, old_value)
```

which would be used in 3-4 places from a quick look. Thoughts?

Reply:
I have reviewed all ``finally`` in our code, I am not sure it is worth the abstraction at this point, but we can discussed, I am not opposed.